### PR TITLE
Nano: Support passing only x for keras's `optimize`/`quantize`

### PR DESCRIPTION
**Pull request by rnwang04**
## Description

Support user passing only x without y in InfererenceOptimizer's `quantize` / `optimize` in keras.

### 1. Why the change?

https://github.com/analytics-zoo/nano/issues/242#event-8042561418 issue 1

### 2. User API changes

No change, but now user can only pass tensor/numpy (tuple) input x like:
```
model = ResNet50(weights=None, input_shape=[224, 224, 3], classes=10)
model = Model(inputs=model.inputs, outputs=model.outputs)
input_examples = np.random.random((100, 224, 224, 3))
# for quantize
model = InferenceOptimizer.quantize(model,
                                    accelerator='openvino',
                                    x=input_examples)
# for optimize
model = InferenceOptimizer.optimize(model,
                                    x=input_examples)
```

### 3. Summary of the change 

- Support user passing only x without y in InfererenceOptimizer's `quantize` / `optimize` in keras.
- related uts

### 4. How to test?
- [ ] Unit test

### CODE REVIEW

File: python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py [hunk 1]
```diff
@@ -430,9 +424,9 @@ def quantize(model: Model,
         :return:            A TensorflowBaseModel for INC. If there is no model found, return None.
         """
         invalidInputError(approach == 'static', "Only 'static' approach is supported now.")
-        invalidInputError(y is not None or isinstance(x, tf.data.Dataset),
-                          "y can be omitted only when x is a Dataset which returns \
-                              tuples of (inputs, targets)")
+        if not isinstance(x, tf.data.Dataset) and y is None:
```
Comment (MeouSker77) [inline python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py:427]: may also handle the cases that `x` is a `Dataset` but only return feature no label

**Review by MeouSker77 — APPROVED** [review]
others LGTM

File: python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py [hunk 2]
```diff
@@ -430,9 +423,16 @@ def quantize(model: Model,
...
+            # fake label to make quantization work
+            y = range(len(x))
+        # for TensorDataset
+        from tensorflow.python.data.ops.dataset_ops import TensorDataset, TensorSliceDataset
```
Comment (MeouSker77) [inline python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py:430]: may change to:

```python
batch = next(iter(x))
if not isinstance(batch, tuple) or len(batch) == 1:
    y = tf.data.Dataset.from_tensor_slices(list(y))    # this `list(y)` is necessary
    x = tf.data.Dataset.zip(x, y)
```

It can handle more cases, not only `TensorDataset` or `TensorSliceDataset`

**Review by MeouSker77 — APPROVED** [review]

**Comment by MeouSker77** [comment]
others LGTM

**Review by MeouSker77 — APPROVED** [review]
LGTM
